batch/batch/server/server.py

```diff
@@ -791,35 +745,54 @@ def from_record(record):
                                        ttl=ttl,
                                        is_open=is_open,
                                        userdata=json.dumps(userdata),
-                                       user=user)
+                                       user=user,
+                                       deleted=False)
 
         batch = Batch(id=id, attributes=attributes, callback=callback,
-                      ttl=ttl, is_open=is_open, userdata=userdata, user=user)
+                      ttl=ttl, is_open=is_open, userdata=userdata, user=user,
+                      state='running', complete=False, deleted=False)
         batch.close_ttl()
         return batch
 
-    def __init__(self, id, attributes, callback, ttl, is_open, userdata, user):
+    def __init__(self, id, attributes, callback, ttl, is_open, userdata, user,
+                 state, complete, deleted):
         self.id = id
         self.attributes = attributes
         self.callback = callback
         self.ttl = ttl
         self.is_open = is_open
         self.userdata = userdata
         self.user = user
+        self.state = state
+        self.complete = complete
+        self.deleted = deleted
 
     async def get_jobs(self):
         return [Job.from_record(record) for record in await db.jobs.get_records_by_batch(self.id)]
 
     async def cancel(self):
+        await self.close()
         jobs = await self.get_jobs()
         for j in jobs:
             await j.cancel()
 
+    async def mark_deleted(self):
+        await self.close()
+        await db.batch.update_record(self.id,
+                                     deleted=True)
+        self.is_open = False
```

Comment:
This is already done by close.  This code is duplicating the logic in cancel, and I think can just be:

```
  self.cancel()
  await db.batch.update_record(self.id, deleted=True)
  self.deleted = True
```